src/buffer.jl

```diff
@@ -38,7 +38,13 @@ Base.seek(buffer::FileBuffer, n::Integer) = seek(buffer.io, n)
 Base.seekstart(buffer::FileBuffer) = seekstart(buffer.io)
 Base.seekend(buffer::FileBuffer) = seekend(buffer.io)
 Base.position(buffer::FileBuffer) = position(buffer.io)
-Base.eof(buffer::FileBuffer) = eof(buffer.io)
+function Base.eof(buffer::FileBuffer)
```

Comment:
I guess the reason pre-emptively reading the file before calling `eof` is fine is because our `_read` function checks the size of the `io` first... so this should only happen once?

Reply:
yes, that's my understanding